### ISSUE TITLE
Primary logical models should include all oncology.txt elements

### ISSUE THREAD

**Issue by rutemartins** [standardhealth/ballot#19]
The primary logical models table does not contain all breast-cancer specific data elements (e.g. it's missing result components for HER2byISH). However, it includes similar supporting elements such as BreastSite and BreastSpecimen. I understand what determines the inclusion of elements in the primary table is whether they are defined as EntryElements, but the distinction from a conceptual standpoint, is confusing. Can we have all elements defined/constrained in oncology.txt in the primary table?

**Comment by cmoesel** [comment]
@markkramerus will have to confirm if he wants this.  We discussed it last week and he said he was going to make the entry/non-entry thing work.

The problem with the idea of a primary *namespace* is that not all IGs will have a primary namespace (or may have more than one).  For example, the IG we make for all of SHR -- what would you consider its primary namespace to be?

That said, I may be able to develop a configurable "primary element strategy" that allows authors to determine if primary elements should be determined by Entry/Non-Entry or by membership in a namespace -- but we'll have to determine if such a feature should be my priority right now.

**Comment by markkramerus** [comment]
I noticed this too, and for clarity, I can change the oncology elements to EntryElements to boost them to the top table, even though components shouldn't be entries.

**Comment by markkramerus** [comment]
On the other hand, I don't want to generate profiles for components.  Leaving as is.

**Comment by cmoesel** [comment]
I've implemented the concept of primary selection strategies.  The default strategy is the "entry" strategy, which makes all entries "primary" profiles and models.

I've added a new "namespace" strategy that allows you to say that any elements in a list of namespaces should be considered "primary".  For example, if we want all `oncology` elements to be in the "primary" listing of profiles and models, we'd add this to the config.json:
```
    "igPrimarySelectionStrategy": {
        "strategy": "namespace",
        "primary": ["oncology"]
    },
```
